### ISSUE TITLE
Unlisted posts with hashtags should show up only in the hashtag feed and the person's profile, but not main feeds

### ISSUE THREAD

**Issue by githubgirly** [mastodon/mastodon#23038]
### Pitch

The point of an unlisted post is to be unlisted from the main feeds, federated and local, so when somebody uses a hashtag on an unlisted post it should exclude the main feeds but still show up in the hashtag's specific feed as well as the user's profile. The purpose of a hashtag is to offer discoverability based specifically on searching that term, and the main appeal of unlisted posts is to avoid showing up in the federated and local tabs which is just from clicking on that tab, so unlisted posts with hashtags should show up in those feeds

### Motivation

This would be helpful from those who want to avoid the noise of random people from the local feed and just have an audience specifically with the interests of the hashtag since they searched it up

**Comment by timtfj** [comment]
I would find this useful on occasion.

I suppose an issue is that it could turn up a lot of out-of-context posts from the middle of threads, which might not be wanted. But this could be solved by adding a "Public posts only" switch to the hashtag display, so people can choose whether to see unlisted ones or not.

**Comment by juliloving** [comment]
> I would find this useful on occasion.
> 
> I suppose an issue is that it could turn up a lot of out-of-context posts from the middle of threads, which might not be wanted. But this could be solved by adding a "Public posts only" switch to the hashtag display, so people can choose whether to see unlisted ones or not.

Maybe unlisted replies won't be counted, since it's just a conversation at that point

**Comment by mdingemanse** [comment]
Very sensible suggestion. Right now hashtags in unlisted posts are entirely useless, which is counterintuitive. We make unlisted posts to avoid cluttering main feeds, but not too avoid being found or read (that's what follower only or private posts are for). So, hashtag feeds should include unlisted posts.

**Comment by simon-frankau** [comment]
To give a concrete example, a friend wanted this exact feature: During Eurovision, they wanted to get a bit spammy with their enthusiastic, somewhat obscure Eurovision posts, which would have been perfectly at home on #Eurovision or whatever, but that they quite sensibly didn't want to clog up the main feeds with (and probably confuse a bunch of people into the bargain).

Wanting to be visible in the main feed and wanting to be visible to a specific group via hashtags are quite distinct wishes, and I don't think they should be so tightly coupled together.

**Comment by trwnh** [comment]
> The point of an unlisted post is to be unlisted from the main feeds, federated and local

nope -- the "point" of unlisted is literally only to opt out of the public timelines. if you follow someone, you see their posts.

> We make unlisted posts to avoid cluttering main feeds, but not too avoid being found or read (that's what follower only or private posts are for)

this is patently wrong, though -- the current rationale is that you should be able to discuss and link to a hashtag without appearing in that hashtag, and without being forced to post privately.

> Wanting to be visible in the main feed and wanting to be visible to a specific group via hashtags are quite distinct wishes, and I don't think they should be so tightly coupled together.

there is a point here, but i think it's a classic x-vs-y problem. the real problem is that the way hashtags work, they aren't aggregated anywhere specific; and the way posts work, they are always sent to your followers. this has nothing to do with "unlisted", but it could be a separate feature request to have a way to address a post to the public *without* addressing it to followers. (the technical challenge, of course, is that "public" is not a real recipient, so you would need to figure out who you're actually sending your activity to...)

**Comment by markrprior** [comment]
I run a couple of games on the bird site and Mastodon where entries are tagged with a particular hashtag. I don't need to follow these accounts to see their entry, which allows new players to just start playing. If someone is posting a thread that includes their entry as an unlisted post in that thread then it does not appear on the timeline of the game's hashtag. They have to tag me specifically for me to see it and/or I follow them. If I take a night off and someone fills in for me then it is unlikely that they will see these entries. Therefore having posts including a hashtag (which I follow) show up irrespective of whether they are public or unlisted would be good.

**Comment by brendanjones** [comment]
Yup, if you don’t want an unlisted post to show up in hashtag feeds … don’t use a hashtag. Using a hashtag generally implies a desire for discoverability. 

The only exception I can think of is if you’re talking *about * a hashtag but don’t intend for the post to be seen by the tag’s followers. An example might be suggesting the tag to someone else. One could of course get around spamming the hashtag feed by writing the hashtag without putting the actual hash symbol in front of it. Or when markdown formatting is here, hashtags in code blocks shouldn’t be treated as a hashtag. 

The other way to deal with this is separating out post visibility and distribution settings from each other. Right now they’re combined, but we could make one setting where you select to which feeds your post gets sent. This is a bigger change though, both technically and from a UX perspective, and I’d suggest user research validation before making such a change. In the mean time, I think this issue suggests a logical improvement - unlisted posts with hashtags should show up in hashtag feeds.

**Comment by mdingemanse** [comment]
Strongly agree wint @brendanjones that the original issue suggests a simple and logical improvement. Let me add a belated response to @trwnh above, who writes in response to me and others describing what we see people doing and expecting:
>this is patently wrong, though -- the current rationale 

Users don't generally look up 'the current rationale' or the original design idea when using tools and their affordances. From a UX perspective and across social media, discoverability is the primary affordance of hashtags and it is logical for people to expect they work this way. So when the majority of folks in this thread note that they expect and prefer unlisted hashtagged posts to show up in the relevant feeds, 'patently wrong' is not really a relevant or helpful characterization — you're getting information about what actual users do and expect.

In fact, the supposed use case underlying this design choice —being "able to discuss and link to a hashtag without appearing in that hashtag"— appears to be very rare indeed, and easily addressed in the ways @brendanjones describes.

If we see an order of magnitude _more_ people using unlisted-hashtags in the expectation that they'll be discoverable than people using them in the way they are 'intended', then I would argue that is reason to rethink the rationale.

**Comment by trwnh** [comment]
> it is logical for people to expect they work this way. 

This is counter-intuitive. People expect something that goes directly contrary to the purpose of the feature. Asking for unlisted posts to show up in public feeds is like asking for unlisted YouTube videos to show up in search.

> Yup, if you don’t want an unlisted post to show up in hashtag feeds … don’t use a hashtag. Using a hashtag generally implies a desire for discoverability.

This could just as easily be stated in the inverse. "If you want an unlisted post to show up in hashtag feeds... don't use unlisted. Using unlisted generally implies a desire to opt-out of discoverability."

**Comment by theastropath** [comment]
>  Asking for unlisted posts to show up in public feeds is like asking for unlisted YouTube videos to show up in search.

I would say it's more like watching a playlist and expecting unlisted videos to show up within it (Which they do).

**Comment by brendanjones** [comment]
> > it is logical for people to expect they work this way.
> 
> This is counter-intuitive. People expect something that goes directly contrary to the purpose of the feature. Asking for unlisted posts to show up in public feeds is like asking for unlisted YouTube videos to show up in search.
> 

It's not counter-intuitive, because people don't understand what unlisted does. What do they expect it to do?

I ran a poll to find out: https://fosstodon.org/@Brendanjones/110894499551248739 

The result? Only 52% of 85 people understood what unlisted does correctly. That’s shockingly low, and points to the need for a change somewhere. 

A full 33% don’t know what it does at all, though I don’t know if that’s them not understanding only ‘unlisted’, or not knowing about these visibility settings at all, or something else. 

One source of the confusion is that there are two post attributes controlled in this one setting. It’s currently not clear how each setting affects each attribute.

The attributes:
1. Post privacy: who is *able* to see the post? (i.e. if they have the post link)
2. Post distribution: In which feeds is the post displayed?

Even if we stick with the existing settings, I think they can be improved by:
1. Rename ‘Unlisted’. It is inaccurately named because it implies posts don’t get distributed to any lists, including home, yet this is not the case.
2. The descriptions for each setting need improving, to state the distribution and privacy level of each.
3. The wiki page https://docs.joinmastodon.org/user/posting/#privacy needs to better explain the distribution and privacy level of each setting.

**Comment by trwnh** [comment]
> One source of the confusion is that there are two post attributes controlled in this one setting. It’s currently not clear how each setting affects each attribute [...] I think they can be improved by: 1. Rename 'Unlisted'.

I don't disagree with you; I do think the issue is in communication and less in actual functionality. If it were up to me, I'd change "Unlisted" -> "Public", and what is currently "Public" should be called "Discoverable". This is because from a "privacy" perspective, the scope of both is to show it to anyone who asks for it. The only difference is entirely Mastodon-specific: opting out of public timelines only makes sense if there is a concept of a "public timeline" in the first place. A lot of the confusion comes from this bandaid patch from 7 years ago, because new Mastodon users (2016 era) wanted to opt-out of the firehose and hashtag views, which is an expectation that largely wasn't present in GNU Social before then. Of course, now the difference covers more Mastodon-specific functionality like trends and full-text search. There is a gradual ongoing transition to using specific properties like `discoverable` and `indexable` which are currently exposed on the profile, and I've advocated in the past that this should go further, such as by applying these to posts. (See #12178 or #18212 among other issues.)

It occurs to me that this is also in effect very similar to #21645 or other such issues. God knows there's a lot of issues around "unlisted" and what people expect it to do or not do. Several are mutually exclusive, even!

**Comment by brendanjones** [comment]
> I'd change "Unlisted" -> "Public", and what is currently "Public" should be called "Discoverable".

I agree with the reasoning but I'd avoid using the word 'discoverable' because discoverability is too easily confused with searchability. An unlisted (or 'Public', in your new naming) post will still be discoverable through search. especially with full-text search incoming. I'm not sure of a better name, though.

> I don't disagree with you; I do think the issue is in communication and less in actual functionality.

Well, I think it's both :) Definitely the communication, but I think we either need to 
1. Split out the privacy and distribution settings from each other (I've no idea if that's technically possible, I don't know the specifics of what attributes determine the privacy and distribution), or
2. Add more options to the existing list. 
a. I'd start with adding what this issue calls for: a setting for posts that sends only to mentioned people and hashtags, but is still visible (public) to all. So not sent to home (of followers), local or federated feeds.

**Comment by brendanjones** [comment]
In Mastodon 4.2.0, unlisted posts will be opted-out of full-text search, even if the user has opted in to their posts being searchable in their privacy settings. 

I don’t agree with per-post search opt-in/out being bundled into this setting, but since it is, this is even more needed:

> a setting for posts that sends only to mentioned people and hashtags, but is still visible (public) to all. So not sent to home (of followers), local or federated feeds.

I would make this new post type opted in to search. Otherwise, if you want your posts searchable you always have to post public even if you don’t want to spam timelines.

**Comment by jocelynetienne** [comment]
> Wanting to be visible in the main feed and wanting to be visible to a specific group via hashtags are quite distinct wishes, and I don't think they should be so tightly coupled together.

Indeed. 

One more usage example : I run RSSfeed-based bots such as https://botsin.space/@flypapers and https://botsin.space/@Rxiv_cytoskeleton for the scientific community. I detect keywords from the RSS so people can focus on papers relevant to their research. But, for good reasons, botsin.space now makes most posts unlisted (see https://botsin.space/@muffinista/110378654504428500), which renders these hashtagged posts undetectable by people following hashtags rather than the bots.

**Comment by mdingemanse** [comment]
Revisiting this to retract my earlier support for this and note that I'm coming round to @trwnh's position that unlisted is sufficiently confusing and seeing divergent / conflicting uses that repurposing it as this issue proposes won't solve the trouble or serve the use cases people have brought up. Rather, we need discoverability at post level, for instance as a visibility setting as suggested in #12178 and #21645. I've summarised some of the use cases from across threads in [this comment](https://github.com/mastodon/mastodon/issues/21645#issuecomment-1880717737), and it seems they are best addressed using something that is distinct from unlisted.

**Comment by brendanjones** [comment]
Just adding this here, since this is a duplicate of it, and it's got lots of conversation in it: https://github.com/mastodon/mastodon/issues/5463

**Comment by The-Beebz** [comment]
I also add my +1 for this, because its simply integral for personal management of being able to look back at your profile. I get that the style of the interface is pretty ephemeral in its first intention, but the reality is that a lot of artists especially, myself included. Like to organize our stuff with hashtags even here. Posting unlisted is pretty much only incentivized for replies  but i hate that i cant rediscover something unless its been "shouted" to the main feeds
